### DISCUSSION TITLE
Uploads API issue

### DISCUSSION THREAD

**Discussion by alexluca89** [amzn/selling-partner-api-models#3161] [General]
I am calling the uploads API successfully and I get back the response containing the uploadDestinationId and the pre-signed url.

I am trying to use this pre-signed url to upload my file. I correctly set all the headers and the raw data of the file but this is what I get back from Amazon:

![image](https://user-images.githubusercontent.com/37292947/137748825-ad37f119-ede6-4067-867b-00834eae9f16.png)

Is anyone else facing the same issue?

**Comment by JordanForeman** [comment]
What do you mean when you say that you:

> correctly set all the headers

Are you manually signing the request and adding signature headers? That might be the problem; a _pre-signed_ URL is just that: it's already signed! If that's what you're doing, my guess would be that Amazon is attempting to upload using the signature that _you_ provide in the headers, which doesn't match the headers of the pre-signed operation.

**Comment by alexluca89** [comment]
Sorry I wasn't clear enough. What I mean by setting the correct headers is setting up the correct MD5 and encryption. I do not sign the URL. 
I just get the URL from the response when calling the uploads API and then setting up the headers as in the example below. 

`{
   "x-amz-server-side-encryption": "aws:kms",
   "Content-MD5": "4e8J8PdWZtcotD13saApsQ=="
}`

I use **PUT** request to try to send the raw data to this URL but I get back **Access Denied**.
